toolkit/featureforms/src/main/java/com/arcgismaps/toolkit/featureforms/internal/components/attachment/AttachmentElementState.kt

```diff
@@ -80,7 +82,7 @@ internal class AttachmentElementState(
     /**
      * The attachments associated with the form element.
      */
-    val attachments = SnapshotStateList<FormAttachmentState>()
+    var attachments by mutableStateOf(emptyMap<Int, FormAttachmentState>())
```

Comment:
FormAttachments are unique, persistent, and implement equals and hashCode. You can key on them, and if that doesn't work there is likely a bug somewhere.
```suggestion
    var attachments by mutableStateOf(emptyMap<FormAttachment, FormAttachmentState>())
```

Reply:
I believe this should work too, but it the adds a `FormAttachment` dependency to the `AttachmentFormElement`. This would mean previews cannot be generated with a list of attachments.

```
@Composable
AttachmentFormElement(
    ..
    attachments: Map<FormAttachment, FormAttachmentState>,
    ..
)